lib/Paws/Net/RestXmlCaller.pm

```diff
@@ -79,6 +79,14 @@ package Paws::Net::RestXmlCaller;
     {
       if ($attribute->does('Paws::API::Attribute::Trait::ParamInURI')) {
         my $att_name = $attribute->name;
+        my $non_print = join('',  map { chr($_) } (128..255) );
+        if ($call->$att_name =~ /[{^}`\[\]><#%'"~|\\$non_print]/) {
+          return Paws::Exception->throw(
+            message => "Found unacceptable content in $att_name parameter",
+            code => 'InvalidInput',
+            request_id => '',
+           );
+        }
         if ($uri_attrib_is_greedy{$att_name}) {
             $vars->{ $attribute->uri_name } =  uri_escape_utf8($call->$att_name, q[^A-Za-z0-9\-\._~/]);
```

Comment:
Adding the `=` character fixes a double encoding bug we are encountering. Would there be any perceived harm of including it in the regex, or making the regex configurable or overridable? Would a separate ticket be necessary since it is related to URI encoding?

I can probably find some time to work on this to send a fix and tests.

i.e. Key = `client=TEST/year=2020/month=01/day=02/test.txt`

Reply:
Hi, sorry, I didn't see these comments - this change hasnt been merged yet, however we're working on it. The issue should be fixed in #265 , there's a test in t/s3/uri_encoding.t that checks = is encoded . Are you saying it shouldn't be? Its on the list here: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html

See "Characters that might require special handling" section. (Though the "might" makes for great documentation.. either we do or we don't.. !)